### PR TITLE
add EdgeMesh-and-EdgeCore-decoupling-design

### PR DESCRIPTION
**Pull request by BaiXue007**
Signed-off-by: BaiXue <506874018@qq.com>

**What type of PR is this?**

/kind design

**What this PR does / why we need it**:

EdgeMesh and EdgeCore decoupling design

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# EdgeMesh and EdgeCore decoupling design
+
+## Motivation
+
+```
```
Comment (daixiang0) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:5]: We need it if it is a code field, since  all of them are text, please remove "```"

**Review by fisherxu — COMMENTED** [review]
Need more details about the decouple work.

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+EdgeMesh supports service mesh functions of the edge, including service discovery and traffic management. The current EdgeMesh functions are relatively independent and suitable for decoupling from EdgeCore to deploy as an application container, facilitating the subsequent upgrade and maintenance of EdgeMesh.
+```
```
Comment (kevin-wangzefeng) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:7]:
```suggestion
EdgeMesh supports service mesh functions of the edge, including service discovery and traffic management. The current EdgeMesh functions are relatively independent and suitable for decoupling from EdgeCore to deploy as an application container, facilitating the subsequent upgrade and maintenance of EdgeMesh.
```

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Goals
+
+- ```
+  The EdgeMesh module is decoupled from EdgeCore and supports EdgeMesh to be deployed as an independent application container.
+  ```
+
+- ```
+  The deployment methods also ensure forward compatibility, deployment methods supports users to choose EdgeMesh container deployment or built-in EdgeCore.
+  ```
```
Comment (kevin-wangzefeng) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:17]:
```suggestion
- The EdgeMesh module is decoupled from EdgeCore and supports EdgeMesh to be deployed as an independent application container.

- The deployment methods also ensure forward compatibility, deployment methods supports users to choose EdgeMesh container deployment or built-in EdgeCore.
```

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+- *In the future, the new service mesh features will only be reflected on the EdgeMesh module decoupled from EdgeCore.*
+
+## Proposal
+
+```
+The function of EdgeMesh is relatively independent, and it is suitable for decoupling from EdgeCore and deploying as an application container, which is convenient for subsequent upgrade and maintenance.
+```
```
Comment (kevin-wangzefeng) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:25]:
```suggestion
The function of EdgeMesh is relatively independent, and it is suitable for decoupling from EdgeCore and deploying as an application container, which is convenient for subsequent upgrade and maintenance.
```

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  Support EdgeMesh to be independent of EdgeCore and be deployed as an application container.
+  ```
+
+- ```
+  The deployment methods supports forward compatibility, retains the ability of the enable module EdgeMesh in EdgeCore. Users can choose container deployment or built-in EdgeCore deployment.
+  ```
+
+- ```
+  The installation tool keadm supports users to choose the deployment method of EdgeMesh: EdgeMesh module built in EdgeCore or  is deployed as a container application. By default, the application deployment method is adopted.
+  ```
```
Comment (kevin-wangzefeng) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:41]:
```suggestion
- Support EdgeMesh to be independent of EdgeCore and be deployed as an application container.

- The deployment methods supports forward compatibility, retains the ability of the enable module EdgeMesh in EdgeCore. Users can choose container deployment or built-in EdgeCore deployment.

- The installation tool keadm supports users to choose the deployment method of EdgeMesh: EdgeMesh module built in EdgeCore or  is deployed as a container application. By default, the application deployment method is adopted.
```

File: docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+2、Communication：
+
+- ```
+  If EdgeMesh is deployed as an independent application container, the message channels for the addition, deletion and modification of the "service" service object involved in EdgeMesh will be changed from the original "channel" communication to the "list-watch" communication . For the specific list-watch design, please refer to https://github.com/kubeedge/kubeedge/pull/2508.
+  ```
```
Comment (kevin-wangzefeng) [inline docs/proposals/EdgeMesh-and-EdgeCore-decoupling-design.md:47]:
```suggestion
- If EdgeMesh is deployed as an independent application container, the message channels for the addition, deletion and modification of the "service" service object involved in EdgeMesh will be changed from the original "channel" communication to the "list-watch" communication . For the specific list-watch design, please refer to https://github.com/kubeedge/kubeedge/pull/2508.
```

**Pull request closed without merging**